protected/humhub/modules/comment/widgets/CommentControls.php

```diff
@@ -60,19 +60,36 @@ public function initControls()
         }
 
         if ($this->comment->canDelete()) {
-            $deleteUrl = Url::to(['/comment/comment/delete', 'objectModel' => $this->comment->object_model,
+            $isAdmin = Yii::$app->user->isAdmin() && $this->comment->created_by !== Yii::$app->user->id;
```

Comment:
We should trigger this modal for every user which is allowed to delete comments.
This could be also moderators.